src/Entity/ProductBundle.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+namespace Drupal\commerce_product\Entity\ProductBundle;
+
+use Drupal\commerce\PurchasableEntityInterface;
+use Drupal\commerce_product\Entity\Product;
+use Drupal\Core\Entity\ContentEntityBase;
+use Drupal\Core\Entity\EntityTypeInterface;
+use Drupal\Core\Field\BaseFieldDefinition;
+
+/**
+ * Defines the product entity class.
+ *
+ * @ContentEntityType(
+ *   id = "commerce_product_bundle",
+ *   label = @Translation("Product Bundle"),
+ *   label_singular = @Translation("product bundle"),
+ *   label_plural = @Translation("product bundles"),
+ *   label_count = @PluralTranslation(
+ *     singular = "@count product bundle",
+ *     plural = "@count product bundles",
+ *   ),
+ *   handlers = {
+ *     "event" = "Drupal\commerce_product\Event\ProductEvent",
+ *     "storage" = "Drupal\commerce\CommerceContentEntityStorage",
+ *     "access" = "\Drupal\commerce\EntityAccessControlHandler",
+ *     "permission_provider" = "\Drupal\commerce\EntityPermissionProvider",
+ *     "view_builder" = "Drupal\Core\Entity\EntityViewBuilder",
+ *     "list_builder" = "Drupal\commerce_product\ProductListBuilder",
+ *     "views_data" = "Drupal\views\EntityViewsData",
+ *     "form" = {
+ *       "default" = "Drupal\commerce_product\Form\ProductForm",
+ *       "add" = "Drupal\commerce_product\Form\ProductForm",
+ *       "edit" = "Drupal\commerce_product\Form\ProductForm",
+ *       "delete" = "Drupal\Core\Entity\ContentEntityDeleteForm"
+ *     },
+ *     "route_provider" = {
+ *       "default" = "Drupal\Core\Entity\Routing\AdminHtmlRouteProvider",
+ *       "delete-multiple" = "Drupal\entity\Routing\DeleteMultipleRouteProvider",
+ *     },
+ *     "translation" = "Drupal\commerce_product\ProductTranslationHandler"
```

Comment:
translation handler will not be from commerce_product, but probably from commerce_product_bundle if not this sub-module.